src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs

```diff
@@ -954,5 +954,58 @@ string GetPropertyValue(string propertyName)
 
             GetPropertyValue("RootNamespace").Should().Be("Project_Name_With_Spaces");
         }
+
+        [Fact]
+        public void It_errors_on_windows_sdk_assembly_version_conflicts()
+        {
+            var testProjectA = new TestProject()
+            {
+                Name = "ProjA",
+                TargetFrameworks = "net5.0-windows10.0.19041"
+            };
+            testProjectA.SourceFiles.Add("ProjA.cs", @"namespace ProjA
+{
+    public class ProjAClass
+    {
+        public static string str { get { return Windows.Media.Devices.ColorTemperaturePreset.Auto.ToString(); } }
+        public string ProjBstr { get { return ProjB.ProjBClass.str; } }
+    }
+}");
+            var testProjectB = new TestProject()
+            {
+                Name = "ProjB",
+                TargetFrameworks = "net5.0-windows10.0.19041",
+            };
+            testProjectB.SourceFiles.Add("ProjB.cs", @"namespace ProjB
+{
+    public class ProjBClass
+    {
+        public static string str { get { return Windows.Media.Devices.ColorTemperaturePreset.Auto.ToString(); } }
+    }
+}");
+            testProjectA.ReferencedProjects.Add(testProjectB);
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProjectA)
+                .WithProjectChanges((path, project) =>
+                {
+                    if (Path.GetFileName(path).Equals("ProjA.csproj"))
+                    {
+                        var ns = project.Root.Name.Namespace;
```

Comment:
Suggestion: Add a comment explaining what the project modification is doing
```suggestion
                        //  Use a previous version of the Microsoft.Windows.SDK.NET.Ref package, to
                        //  simulate the scenario where a project is compiling against a library from NuGet
                        //  which was built with a more recent SDK version.
                        var ns = project.Root.Name.Namespace;
```